### ISSUE TITLE
Repository information in  pip.conf(s) is not used

### ISSUE THREAD

**Issue by lfvjimisola** [sam-atkins/pylibyear#11]
[The software](https://github.com/sam-atkins/pylibyear/blob/master/libyear/pypi.py) assumes that pypi.org is the correct repositories but there might be others. Repository information, location (url) as well as auth, in pip.conf should be used to allow more flexibility.

**Comment by lfvjimisola** [comment]
@sam-atkins Any chance that you might support this?

**Comment by sam-atkins** [comment]
@lfvjimisola I hadn't considered this previously. The original (unforked) version assumes all packages are on pypi. So in your case you self-host packages somewhere else? 

Unfortunately in Python there are so many package managers and different configuration files that I'm concerned this requires to check several different files to see if to check pypi or another site. Perhaps a command line option for the user to override the default of pypi with their custom url is the simplest option. What do you think?

**Comment by lfvjimisola** [comment]
@sam-atkins Hi! Back from vacation now.

Yes, self-hosting somewhere else. 

Agreed, python eco system with various package managers and configuration files is a challenge.

I like your solution but simply overriding the url on command line is not enough if the repository requires auth.
The auth could be username/password or just password/token using url-based auth or OAuth (e.g. bearer token).

Not sure what's mostly support by the major repository managers such as

* JFrog: [basic](https://jfrog.com/help/r/jfrog-platform-administration-documentation/basic-authentication) and [OAuth](https://jfrog.com/help/r/jfrog-platform-administration-documentation/authorization-headers)
*[Nexus](https://help.sonatype.com/en/user-tokens.html): basic

* [GitLab](https://docs.gitlab.com/ee/user/packages/pypi_repository/)

Perhaps, one can have a look at how pip does and mimic that.

suggestion of command line options:

 - --url (plain url to repository)
 - --username-password
 - --token
 - --auth-type (url or oauth)

other combinations of username, password and token are possible such as username and password/token.

**Comment by sam-atkins** [comment]
Yes, good point re auth. Unfortunately I won't have time to look at this for the foreseeable future. Feel free to open a PR if you have time and motivation 😄

**Comment by lfvjimisola** [comment]
I completely understand. Python is not our main ecosystem and we have some other things that have higher priority for now (such as dynamic versioning for Maven projects). I have an internal ticket on this matter and will, time permitting, follow-up on it later unless you found the time 😊
